### DISCUSSION TITLE
Query over API

### DISCUSSION THREAD

**Discussion by kristofferlindvall** [datalust/seq-tickets#1683] [Help] [answered]
I'd like to create some more elaborate dashboards with graphs not available in Seq's dashboards. While it would be awesome to be able to code custom charts for Seq dashboards, I figured perhaps I could simply use queries over the API and use any of the many dashboarding apps available online instead. But I can't find any documentation around executing queries over API. It only seems possible to get raw event streams. Am I missing something or is it not possible?

**Comment by liammclennan** [comment, marked as answer]
Hi @kristofferlindvall ,

You want the `api/data` endpoint. See https://docs.datalust.co/docs/server-http-api#apidata.

One way to experiment with the API is to run your query in Seq and watch the network tab in the browser dev tools. 

You can do things like:

`POST http://seq.example.com/api/data?q=select count(*) from stream group by @Level&rangeStartUtc=2022-11-08T11:00:00.000Z&timeoutMS=600000`

which returns  a result like:

```
{
    "Columns": [
        "@Level",
        "(count)"
    ],
    "Rows": [
        [
            "Information",
            373
        ],
        [
            "Warning",
            250
        ]
    ],
    "Statistics": {
        "ElapsedMilliseconds": 1.7629
    }
}
```
